util/timers/monotonic.go

```diff
@@ -26,7 +26,7 @@ import (
 // Monotonic uses the system's monotonic clock to emit timeouts.
 type Monotonic struct {
 	zero     time.Time
-	timeouts map[time.Duration]<-chan time.Time
+	timeouts map[TimeoutType]map[time.Duration]<-chan time.Time
```

Comment:
mb array of size `filterTypeLength = 3` instead of map?

Reply:
I think we also need to test whether a deadline was already set for a certain type when calling TimeoutAt. So making this an array makes the check less obvious.